### PR TITLE
AI mention fixes, AI shells can now toggle AI mentions

### PR DESCRIPTION
**Pull request by toomykins**
Since I imagine AI's can get distracted easily while in their shell, Shells now get AI mentions too.

:cl:
 * rscadd: AI Shells now have AI mentions

### CODE REVIEW

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -269,13 +269,15 @@ var/list/headset_modes = list(
 			rendered_message = replacetext(rendered_message, T, "<i style='color: red;'>[T]</i>")
 
 	//AI mentions
-	if(istype(src, /mob/living/silicon/ai) && speech.frequency && speech.job != "AI")
+	if((isAI(src)  || isshell(src)) && speech.frequency && !(findtext(speech.job,"AI")<>0) && (speech.name != src.name))
```
Comment (DamianX) [inline code/modules/mob/living/say.dm]: how does this compile? the `<>` part
did you test this?

Comment (toomykins) [reply code/modules/mob/living/say.dm]: yep. findtext returns 0 if it finds nothing, and something else if something was found. it's essentially != 0

**Comment by gurfan** [comment]
thank you I was having trouble figuring out how to implement this

also - does the AI Private channel tag cause the mention to activate?

**Comment by toomykins** [comment]
> Does the AI Private channel tag cause the mention to activate?

No it does not. if it's wanted i'll edit the code so it does.

File: code/modules/mob/living/say.dm (hunk 1 above)
Comment (ghost) [inline code/modules/mob/living/say.dm]: Wait no why is this necessary? !findtext(speech.job, "AI") should do the same thing as !(findtext(speech.job, "AI")<>0). !var returns true if var is 0 or null. Additionally, <> is identical to the != operator and shouldn't be used for the sake of consistency.

Comment (toomykins) [reply code/modules/mob/living/say.dm]: ye, id sworn i had issues with it that way earlier. committed.

**Review by ghost — APPROVED** [review]
